ESPHome version and Wifi Info text sensor cannot have state_topic changed

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
<!--
Please provide details about your environment below this line. -->
Command line.

**ESP (ESP32/ESP8266, Board/Sonoff):**
<!--
Please provide details about which ESP you're using below.
-->
Esp8266

**Affected component:**
<!--
Please add the link to the documentation at https://esphome.io/index.html of the component in question.
-->
ESPHome Version and Wifi Info text sensors

**Description of problem:**
The ESPHome version and Wifi Info text sensors cannont have their state_topic changed, it causes a compilation error.

**Problem-relevant YAML-configuration entries:**
```yaml
text_sensor:
  - platform: version
    name: ESPHome Version
    id: version
    state_topic: devices/habitat_chambre_alex.local/esphome_version
  - platform: wifi_info
    ip_address:
      name: WiFi IP Address
      id: wifi_ip_address
      state_topic: devices/habitat_chambre_alex.local/wifi_ip
    ssid:
      name: Wifi SSID
      id: wifi_ssid
      state_topic: devices/habitat_chambre_alex.local/wifi_ssid
```

**Traceback (if applicable):**
<!--
Please copy the traceback here if compilation is failing. If possible, also connect to the ESP over USB and copy its logs into the backticks.
-->
```
Compiling .pioenvs\habitat_chambre_alex\src\main.cpp.o
src\main.cpp: In function 'void setup()':
src\main.cpp:67:23: error: request for member 'set_custom_state_topic' in 'version->esphome::text_sensor::VersionTextSensor::<anonymous>.esphome::text_sensor::TextSensor::get_mqtt()', which is of pointer type 'esphome::text_sensor::MQTTTextSensor*' (maybe you meant to use '->' ?)
version->get_mqtt().set_custom_state_topic("devices/habitat_chambre_alex.local/esphome_version");
^
src\main.cpp:68:31: error: request for member 'set_custom_state_topic' in 'wifi_ip_address->esphome::text_sensor::IPAddressWiFiInfo::<anonymous>.esphome::text_sensor::TextSensor::get_mqtt()', which is of pointer type 'esphome::text_sensor::MQTTTextSensor*' (maybe you meant to use '->' ?)
wifi_ip_address->get_mqtt().set_custom_state_topic("devices/habitat_chambre_alex.local/wifi_ip");
^
src\main.cpp:69:25: error: request for member 'set_custom_state_topic' in 'wifi_ssid->esphome::text_sensor::SSIDWiFiInfo::<anonymous>.esphome::text_sensor::TextSensor::get_mqtt()', which is of pointer type 'esphome::text_sensor::MQTTTextSensor*' (maybe you meant to use '->' ?)
wifi_ssid->get_mqtt().set_custom_state_topic("devices/habitat_chambre_alex.local/wifi_ssid");
^
*** [.pioenvs\habitat_chambre_alex\src\main.cpp.o] Error 1
```

**Additional information and things you've tried:**




<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

--- a/esphome/components/text_sensor/__init__.py
+++ b/esphome/components/text_sensor/__init__.py
@@ -46,7 +46,7 @@ def setup_text_sensor_core_(text_sensor_var, config):
         trigger = Pvariable(conf[CONF_TRIGGER_ID], rhs)
         automation.build_automations(trigger, [(std_string, 'x')], conf)
 
-    setup_mqtt_component(text_sensor_var.get_mqtt(), config)
+    setup_mqtt_component(text_sensor_var.Pget_mqtt(), config)
 
 
 def setup_text_sensor(text_sensor_obj, config):

